packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/JSTypeConverterHelper.kt

```diff
@@ -0,0 +1,148 @@
+    result.putGeneric(key, convertedValue)
+  }
+
+  return result
+}
+
+fun <K, V> Map<K, V>.toJSValue(containerProvider: JSTypeConverter.ContainerProvider): WritableMap {
+  val result = containerProvider.createMap()
+
+  for ((key, value) in entries) {
+    val convertedValue = JSTypeConverter.convertToJSValue(value, containerProvider)
+    result.putGeneric(key.toString(), convertedValue)
+  }
+
+  return result
+}
+
+fun <T> Iterable<T>.toJSValue(containerProvider: JSTypeConverter.ContainerProvider): WritableArray {
+  val result = containerProvider.createArray()
+
+  for (value in this) {
+    val convertedValue = JSTypeConverter.convertToJSValue(value, containerProvider)
+    result.putGeneric(convertedValue)
+  }
+
+  return result
+}
+
+fun <T> Array<T>.toJSValue(containerProvider: JSTypeConverter.ContainerProvider): WritableArray {
+  val result = containerProvider.createArray()
+
+  for (value in this) {
+    val convertedValue = JSTypeConverter.convertToJSValue(value, containerProvider)
+    result.putGeneric(convertedValue)
+  }
+
+  return result
+}
+
+fun IntArray.toJSValue(containerProvider: JSTypeConverter.ContainerProvider): WritableArray {
+  return containerProvider.createArray().also {
+    for (value in this) {
+      it.pushInt(value)
+    }
+  }
+}
+
+fun FloatArray.toJSValue(containerProvider: JSTypeConverter.ContainerProvider): WritableArray {
+  return containerProvider.createArray().also {
+    for (value in this) {
+      it.pushDouble(value.toDouble())
+    }
+  }
+}
+
+fun DoubleArray.toJSValue(containerProvider: JSTypeConverter.ContainerProvider): WritableArray {
+  return containerProvider.createArray().also {
+    for (value in this) {
+      it.pushDouble(value)
+    }
+  }
+}
+
+fun Enum<*>.toJSValue(): Any? {
+  val primaryConstructor = requireNotNull(this::class.primaryConstructor) {
+    "Cannot convert enum without the primary constructor to js value"
+  }
+
+  if (primaryConstructor.parameters.isEmpty()) {
+    return this.name
+  } else if (primaryConstructor.parameters.size == 1) {
+    val parameterName = primaryConstructor.parameters.first().name!!
+    @Suppress("UNCHECKED_CAST")
+    val parameterProperty = this::class.declaredMemberProperties
+      .find { it.name == parameterName } as KProperty1<Enum<*>, *>
+
+    return parameterProperty.get(this)
+  }
+
+  throw IllegalStateException("Enum '$javaClass' cannot be used as return type (incompatible with JS)")
+}
+
+internal fun WritableMap.putGeneric(key: String, value: Any?) {
+  when (value) {
+    null, is Unit -> putNull(key)
+    is ReadableArray -> putArray(key, value)
+    is ReadableMap -> putMap(key, value)
+    is String -> putString(key, value)
+    is Int -> putInt(key, value)
+    is Number -> putDouble(key, value.toDouble())
+    is Boolean -> putBoolean(key, value)
+    else -> throw IllegalArgumentException("Could not put ${value.javaClass} to 'WritableMap'")
```

Comment:
```suggestion
    else -> throw IllegalArgumentException("Could not put '${value.javaClass}' to 'WritableMap'")
```

Reply:
or 👇 
```suggestion
    else -> throw IllegalArgumentException("Could not put '${value.javaClass}' to WritableMap")
```

The general rule is to quote dynamic/user's input strings.